modules/wifi-ssid/android/src/main/java/expo/modules/wifissid/WifiSsidModule.kt

```
@@ -0,0 +1,109 @@
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+      try {
+        val cm = appContext.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
+        val caps = cm?.let { manager ->
+          manager.activeNetwork?.let { net -> manager.getNetworkCapabilities(net) }
+        }
+        connectedToWifi = caps?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true
+        if (hasFineLocationPermission()) {
+          ssid = (caps?.transportInfo as? WifiInfo)?.ssid?.sanitize()
+        }
+      } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)